### PR TITLE
Announcer API: Shared controller and HTTP handlers (+ tests)

### PR DESCRIPTION
**Pull request by mdittmer**
## Description

Shared state controller (owner of `Announcer` instance) and handlers for public announcer API.

### CODE REVIEW

**Comment by mdittmer** [comment]
Note: https://travis-ci.org/web-platform-tests/wpt.fyi/jobs/386222196#L7402 is not a problem with the code or tests.

**Comment by lukebjerring** [comment]
Can you rebase on master so the behaviour of Travis is current (and probably passing)?

File: revisions/api/handlers/latest.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		At:    now,
+		Start: now.Add(-2 * epochs[0].GetData().MaxDuration),
+	})
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(a.ErrorJSON(err.Error()))
+		return
+	}
+
+	response, err := api.LatestFromEpochs(revs)
```
Comment (lukebjerring) [inline revisions/api/handlers/latest.go]: nit: One-line the `if`, and use `http.Error` (might need a `[]byte` cast)
    if response, err := ...; err != nil {
        http.Error(w, a.ErrorJSON(err.Error()), http.StatusInternalServerError)
        return
    }

File: revisions/api/handlers/list.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	if len(epochs) == 0 {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(a.ErrorJSON("No epochs"))
+		return
+	}
+
+	q := r.URL.Query()
+
+	numRevisions := 1
+	if nr, ok := q["num_revisions"]; ok {
```
Comment (lukebjerring) [inline revisions/api/handlers/list.go]: - `Query` has a `Get("num_revisions")` method which returns an empty string if missing.
- Probably just parse as an int, then assert size limits.
- Move the `num_revisions` param parsing to the `params.go` file

File: revisions/api/handlers/latest.go (hunk 1 above)
Comment (mdittmer) [reply revisions/api/handlers/latest.go]: Done (push forthcoming).

File: revisions/api/handlers/list.go (hunk 2 above)
Comment (mdittmer) [reply revisions/api/handlers/list.go]: I don't quite follow on this one. Let's sync offline.

Comment (lukebjerring) [reply revisions/api/handlers/list.go]: Ignore the last 2 items there - I misread the `len(1)` assertion as being the number of chars lol.
But yeah, no need to fail for multiple copies - `Get` will only give you the last one.

**Comment by Hexcles** [comment]
Hooray!

**Pull request merged**